include/deal.II/lac/trilinos_tpetra_types.h

```diff
@@ -0,0 +1,127 @@
+// ------------------------------------------------------------------------
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2018 - 2024 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+#ifndef dealii_trilinos_tpetra_types_h
+#define dealii_trilinos_tpetra_types_h
+
+#include <deal.II/base/config.h>
+
+#include "deal.II/base/types.h"
+
+#ifdef DEAL_II_TRILINOS_WITH_TPETRA
+
+#  include <Tpetra_Details_DefaultTypes.hpp>
+
+// Forward declarations
+#  ifndef DOXYGEN
+namespace Tpetra
+{
+  template <class LocalOrdinal, class GlobalOrdinal, class Node>
+  class Import;
+
+  template <class LocalOrdinal, class GlobalOrdinal, class Node>
+  class Export;
```

Comment:
These are all names we don't control -- you are forward-declaring things that are declared in Trilinos. Do we need to worry about these types acquiring more template arguments, for example?

Reply:
Perhaps a better approach would be to just `#include` the headers. That increases compile time, but also seems safer.

Reply:
I had a quick look and there are `*fwd_def.hpp` headers in Trilinos which offer the forward declaration for each of the classes. 
I think that might be the cleanest option, right?

Reply:
Did the replacement, seemed the cleanest solution to while also keeping files small.

Reply:
Great, thank you! This is the best solution indeed.